src/SecTester.Bus/Internal/JsonHttpMethodEnumerationStringConverter.cs

```diff
@@ -1,55 +1,52 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Net.Http;
+using System.Reflection;
+using System.Runtime.CompilerServices;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 
 namespace SecTester.Bus.Internal;
 
 internal class JsonHttpMethodEnumerationStringConverter : JsonConverter<HttpMethod>
 {
-  private static readonly HttpMethod _patch = new("PATCH");
-  private static readonly HttpMethod _copy = new("COPY");
-  private static readonly HttpMethod _link = new("LINK");
-  private static readonly HttpMethod _unlink = new("UNLINK");
-  private static readonly HttpMethod _purge = new("PURGE");
-  private static readonly HttpMethod _lock = new("LOCK");
-  private static readonly HttpMethod _unlock = new("UNLOCK");
-  private static readonly HttpMethod _propfind = new("PROPFIND");
-  private static readonly HttpMethod _view = new("VIEW");
-  private static readonly HttpMethod _trace = new("TRACE");
+  private static readonly IEnumerable<HttpMethod> BaseMethods = typeof(HttpMethod)
+    .GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly)
+    .Where(x => x.PropertyType.IsAssignableFrom(typeof(HttpMethod)))
```

Comment:
`BindingFlags.DeclaredOnly` is responsible for this

```suggestion
```

Reply:
`BindingFlags.DeclaredOnly`  - Specifies that only members declared at the level of the supplied type's hierarchy should be considered. Inherited members are not considered.
Hence it does not perform any type of check and I propose to leave `.Where(x => x.PropertyType.IsAssignableFrom(typeof(HttpMethod)))` just for type casting safety.

Reply:
According to the contract, public static members are instances of `HttpMethod`. It is pointless to check it explicitly.